### PR TITLE
Add a /debug route to NAT MCP frontend to expose MCP tools

### PR DESCRIPTION
**Pull request by zhongxuanwang-nv**
## Description

Closes AIQ-1851

This will be a custom MCP route similar to "/health" and used for displaying tools and their schema without the need for an MCP client.
- **Added**: Public HTTP route GET /debug/tools/list in src/nat/front_ends/mcp/mcp_front_end_plugin_worker.py; supports name (repeatable or comma-separated) and detail flags; returns server_name, count, and tools[] with name, description, is_workflow, and optional schema.  ￼
- **Changed**: MCP front-end wiring to register the new custom route and include schema handling (Pydantic via model_json_schema; ChatRequest simplified to single query field).  ￼
- **Removed**: None
- **Why it matters**: Enables introspecting registered MCP tools and their input schemas without an MCP client—useful for debugging, observability, and integration verification.  ￼
- **Breaking changes / Migrations**: None
- **Risks / Edge cases**: Public debug endpoint may expose tool metadata/schemas; ensure it’s restricted/disabled in production or behind auth; 404 behavior for unknown tools confirmed.  ￼
- **Docs**: Added docs in docs/source/workflows/mcp/mcp-server.md with curl examples and parameter semantics.  ￼

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/NeMo-Agent-Toolkit/blob/develop/docs/source/resources/contributing.md).
- We require that all contributors "sign-off" on their commits. This certifies that the contribution is your original work, or you have rights to submit it under the same license, or a compatible license.
  - Any contribution which contains commits that are not Signed-Off will not be accepted.
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

## Summary by CodeRabbit

- New Features
  - Public debug endpoint /debug/tools/list to inspect available MCP tools; supports filtering by name (repeated or comma-separated), a detail flag to include per-tool input schemas, and returns count, tools, and server name; unknown tool names return 404.

- Documentation
  - Added docs describing the debug endpoint, usage examples for name/detail combinations, and guidance for obtaining detailed tool schemas.

- Tests
  - Added comprehensive tests covering route structure, name/detail behaviors, schema representations, edge cases, and error handling.

### CODE REVIEW

**Comment by zhongxuanwang-nv** [comment]
@coderabbitai  review

**Review by AnuradhaKaruppiah — APPROVED** [review]

File: docs/source/workflows/mcp/mcp-server.md [hunk 1]
```diff
@@ -76,6 +76,44 @@ calculator_divide
...
+
+```bash
+curl -s http://localhost:9901/debug/tools/list | jq
+```
+
+This returns a JSON list of tools with names and descriptions.
+
+You can request one or more specific tools by name. The `name` parameter accepts repeated values or a comma‑separated list. When `name` is provided, detailed schemas are returned by default:
+
+```bash
```
Comment (peterychang) [inline docs/source/workflows/mcp/mcp-server.md:90]: Can you add some sample outputs so we can see what format the outputs should be in?

File: src/nat/front_ends/mcp/mcp_front_end_plugin_worker.py [hunk 2]
```diff
@@ -102,6 +106,103 @@ def _get_all_functions(self, workflow: Workflow) -> dict[str, Function]:
...
+    def _setup_debug_endpoints(self, mcp: FastMCP, functions: Mapping[str, FunctionBase]) -> None:
+        """Set up HTTP debug endpoints for introspecting tools and schemas.
+
+        Exposes:
+          - GET /debug/tools/list: List tools. Optional query param `name` (one or more, repeatable or comma separated)
+            selects a subset and returns details for those tools.
+        """
+
+        @mcp.custom_route("/debug/tools/list", methods=["GET"])
+        async def list_tools(request: Request):
```
Comment (peterychang) [inline src/nat/front_ends/mcp/mcp_front_end_plugin_worker.py:118]: Should this function specify the return type?

File: tests/nat/front_ends/mcp/test_mcp_debug_routes.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+            data = resp.json()
+
+            for tool in data["tools"]:
+                assert "is_workflow" in tool  # Always present
+                if expected_schema:
+                    assert "schema" in tool
+                else:
+                    assert "schema" not in tool
+
+    @pytest.mark.asyncio
```
Comment (peterychang) [inline tests/nat/front_ends/mcp/test_mcp_debug_routes.py:202]: This should no longer be necessary. Maybe test to verify and remove if thats the case here
